surefire-providers/surefire-junit-platform/src/main/java/org/apache/maven/surefire/junitplatform/LauncherAdapter.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.maven.surefire.junitplatform;
+
+import java.lang.reflect.Method;
+import java.util.function.Supplier;
+
+import org.apache.maven.surefire.api.util.ReflectionUtils;
+import org.junit.platform.engine.ExecutionRequest;
+import org.junit.platform.launcher.Launcher;
+import org.junit.platform.launcher.LauncherDiscoveryRequest;
+import org.junit.platform.launcher.TestExecutionListener;
+import org.junit.platform.launcher.TestPlan;
+
+import static java.util.Objects.requireNonNull;
+
+final class LauncherAdapter {
+
+    static final Class<?> LAUNCHER_EXECUTION_REQUEST_CLASS = loadLauncherExecutionRequestClass();
+    static final Class<?> CANCELLATION_TOKEN_CLASS = loadCancellationTokenClass();
+    static final Supplier<CancellationTokenAdapter> CANCELLATION_TOKEN_FACTORY = createCancellationTokenFactory();
+    static final Method EXECUTE_METHOD_WITH_LAUNCHER_EXECUTION_REQUEST_PARAMETER =
+            findExecuteMethodWithLauncherExecutionRequestParameter();
```

Comment:
This class heavily relies on reflection to keep compatible with pre-6.x versions of JUnit